src/View/Helper/UrlHelper.php

```diff
@@ -16,15 +16,62 @@
  */
 namespace Cake\View\Helper;
 
+use Cake\Core\App;
+use Cake\Core\Exception\CakeException;
 use Cake\Routing\Asset;
 use Cake\Routing\Router;
 use Cake\View\Helper;
+use Cake\View\View;
 
 /**
  * UrlHelper class for generating URLs.
  */
 class UrlHelper extends Helper
 {
+    /**
+     * Default config for this class
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'engine' => Asset::class,
+    ];
+
+    /**
+     * Asset engine class name
+     *
+     * @var string
+     * @psalm-var class-string<\Cake\Routing\Asset>
+     */
+    protected $_engineClass;
+
+    /**
+     * Constructor
+     *
+     * ### Settings:
+     *
+     * - `engine` Class name to use to replace asset url building functionality.
+     *            The class needs to be placed in the `Routing` directory.
+     *
+     * @param \Cake\View\View $view the view object the helper is attached to.
+     * @param array $config Settings array Settings array
+     * @throws \Cake\Core\Exception\CakeException when the engine class could not be found.
+     */
+    public function __construct(View $view, array $config = [])
+    {
```

Comment:
Instead of overriding the constructor use the initialize() hook. The overriding in TextHelper predates the addition of initialize() hook for helpers.

Reply:
Updated.